File: src/main/java/com/my/admin/app/commerce/model/document/Turn.java
package com.my.admin.app.commerce.model.document;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.io.Serializable;
import java.time.ZonedDateTime;

@Document(collection = "turns")
@Data
public class Turn implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    private Long id;

    private String name;

    private String description;

    private ZonedDateTime startDate;

    private ZonedDateTime endDate;
}

File: src/main/kotlin/com/bk/reactive/app/commerce/admin/my/model/document/Turn.kt
package com.bk.reactive.app.commerce.admin.my.model.document

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.ZonedDateTime

@Document(collection = "turns")
data class Turn(
    @Id
    val id: String,
    val name: String,
    val description: String,
    val startDate: ZonedDateTime,
    val endDate: ZonedDateTime
)